src/Raven.Client/Documents/Conventions/DocumentConventions.cs

```diff
@@ -246,6 +249,21 @@ public Action<JsonSerializer> CustomizeJsonSerializer
             }
         }
 
+        /// <summary>
+        ///     If 'true' Id field will not be added to dynamic objects.
```

Comment:
```suggestion
        ///     By default, the field 'Id' field will be added to dynamic objects, this allows to disable this behavior.
```

That said, let's avoid the double negative here.
Make the convention: "AddIdFieldToDynamicObjects" and default to true

Reply:
done